src/main/java/com/ldtteam/structurize/items/ItemCommandTool.java

```diff
@@ -0,0 +1,464 @@
+            {
+                anchor = BlockPosArgument.getSpawnablePos(cmdContext, ScanCommand.ANCHOR_POS);
+            }
+            String name = "";
+            if (parsed.getContext().getArguments().containsKey(ScanCommand.FILE_NAME))
+            {
+                name = StringArgumentType.getString(cmdContext, ScanCommand.FILE_NAME);
+            }
+
+            setData(context.getItemInHand(), context.getLevel(), pos, getTeleportPos(from, to), name);
+
+            ItemScanTool.setBounds(scanTool, from, to);
+            ItemScanTool.setAnchorPos(scanTool, anchor);
+            Network.getNetwork().sendToPlayer(new ShowScanMessage(from, to, anchor, name), (ServerPlayer) player);
+
+            player.sendMessage(new TranslatableComponent("com.ldtteam.structurize.gui.commandtool.copycommand.ok", name), player.getUUID());
+            player.playNotifySound(SoundEvents.NOTE_BLOCK_CHIME, SoundSource.PLAYERS, 1.0F, 1.0F);
+            return InteractionResult.SUCCESS;
+        }
+        catch (CommandSyntaxException e)
+        {
+            player.sendMessage(new TranslatableComponent("com.ldtteam.structurize.gui.commandtool.copycommand.notscan"), player.getUUID());
+            return InteractionResult.FAIL;
+        }
+    }
+
+    /**
+     * Searches the player's hotbar (only) for a scan tool.
+     * If there's multiple, it prefers the closest item to the current slot, starting from the left side.
+     * @param player The player to search.
+     * @return The scan tool found, or an empty stack.
+     */
+    @NotNull
+    private ItemStack findScanTool(@NotNull final Player player)
+    {
+        final int current = player.getInventory().selected;
+        for (int i = 1; i < 9; ++i)
+        {
+            for (int j = -1; j <= 1; j += 2)
+            {
+                final int slot = current + i * j;
+                if (slot < 0 || slot > 9) continue;
+
+                final ItemStack stack = player.getInventory().getItem(slot);
+                if (stack.is(ModItems.scanTool.get()))
+                {
+                    return stack;
+                }
+            }
+        }
+        return ItemStack.EMPTY;
+    }
+
+    /**
+     * Determines the best position to teleport to, near the given scan bounds.
+     * This can't assume that the area is loaded, so it has to guess something appropriate.
+     *
+     * It's usually most convenient to appear outside the scanned area (it might be filled with placeholders),
+     * so we just pick an arbitrary location on a particular side that's convenient.  Perhaps this
+     * could be made into a setting.
+     *
+     * @param from The first bounding corner.
+     * @param to The second bounding corner.
+     * @return A convenient teleportation position near but not inside the scan area.
+     */
+    @NotNull
+    private static BlockPos getTeleportPos(@NotNull final BlockPos from,

```

Comment:
Seems sus, can we just use the mechanism for finding spawn or sth like that (there are/were many things for finding possible spawn pos in vanilla)

Reply:
I don't want a random position vaguely around the build area, I want a very specific point at at specific side of the build area (where the signs are located).

Reply:
Yeah, you can use this result as center of search area, but I don't see any check that this blockpos is useable as destination = at least two vertical air block hole